ngs_mapper/tests/test_base_caller.py

```diff
@@ -860,10 +835,134 @@ def test_depth_lt_mind(self, mpilecol):
+            'AT': {'baseq': 20*[40]}, # T 21%
+            'AC': {'baseq': 20*[40]}, # C 21%
+        }
+        self.tst_stats(mpilecol, base_stats, ('A',100))
+
+    def test_single_base_insertion_under_80pct(self, mpilecol):
+        base_stats = {
+            'A-': {'baseq': 20*[40]}, # - 21%, not considered for T/C calling
+            'A':  {'baseq': 5*[40]},  # Removed below minth
+            'AT': {'baseq': 55*[40]}, # T 73%
+            'AC': {'baseq': 20*[40]}, # C 27%
+        }
+        self.tst_stats(mpilecol, base_stats, ('AY', 55)) # CT == Y
+
+    def test_single_base_insertion_over_80pct(self, mpilecol):
+        base_stats = {
+            'A-': {'baseq': 1*[40]},  # Removed below minth
+            'A':  {'baseq': 1*[40]},  # Removed below minth
+            'AT': {'baseq': 78*[40]}, # T 86%
+            'AC': {'baseq': 20*[40]}, # C 14%, removed 2nd call below 50%
+        }
+        self.tst_stats(mpilecol, base_stats, ('AT', 78))
+
+    def test_single_base_insertion_ambiguous_base(self, mpilecol):
+        base_stats = {
+            'A-': {'baseq': 48*[40]}, # - 50%
+            'A':  {'baseq': 4*[40]},  # Removed below minth
+            'AT': {'baseq': 48*[40]}, # T 50%
+        }
+        self.tst_stats(mpilecol, base_stats, ('AN', 96))
+
+    def test_insertion_at_low_depth_no_majority(self, mpileupcol):
+        base_stats = {
+            # Ends up, 2 A-, 1 A, 4 AC due to mark_lq biasing reference of 'C'
+            'A-': {'baseq': [40,40,1,1]}, # A- 29% , - 33%
+            'A':  {'baseq': 1*[40]}, # A 14%, removed below minth
+            'AC': {'baseq': [40,40,1,1]}  # AC 57%, C 66%
+        }
+        self.tst_stats(mpilecol, base_stats, ('AC',6))
+
+    def test_insertion_multiple_bases(self, mpileupcol):
+        base_stats = {
+            'AAAA': {'baseq': [40]*20},
+            'AAA-': {'baseq': [40]*20},
+            'AA--': {'baseq': [40]*20},
+            'A---': {'baseq': [40]*20},
+            'A':    {'baseq': [40]*20}
+        }
+        self.tst_stats(mpilecol, base_stats, ('AAA',80))
+
+    def test_insertion_multiple_bases_ambiguous(self, mpileupcol):
+        base_stats = {
+            'AAAA': {'baseq': [40]*100},
+            'AAA-': {'baseq': [40]*100},
+            'AA--': {'baseq': [40]*100},
+            'A---': {'baseq': [40]*100},
+            'A':    {'baseq': [40]*100}
+            'CCCC': {'baseq': [40]*100},
+            'CCC-': {'baseq': [40]*100},
+            'CC--': {'baseq': [40]*100},
+            'C---': {'baseq': [40]*100},
+            'C':    {'baseq': [40]*100}
+        }
+        self.tst_stats(mpilecol, base_stats, ('MMM',80))
+
+    def test_insertion_and_deletion(self, mpileupcol):
+        base_stats = {
+            '*':  {'baseq': [40]*25},
+            '*-': {'baseq': [40]*25},
+            'A':  {'baseq': [40]*25},
+            'AC': {'baseq': [40]*25},
+        }
+        self.tst_stats(mpilecol, base_stats, ('NN', 50))
```

Comment:
Yes it is NN because 50/50 for first and second position. It would be NC if the C was >= 51% as the `-` would be ignored